### ISSUE TITLE
Strange behavior of `without` method

### ISSUE THREAD

**Issue by Marat-Tim** [objectionary/lints#586]
I’m not happy with how the `without()` method works:

1. If I call `without()` twice, only the last call has any effect - previous ones are ignored, which can be misleading
```java
new Source(
    new EoSyntax(
        """
        +architect yegor256@gmail.com
        +home https://www.eolang.org
        +package canonical
        +version 0.0.0
        +spdx SPDX-FileCopyrightText Copyright (c) 2016-2025 Objectionary.com
        +spdx SPDX-License-Identifier: MIT
        
        # Times table, which is a canonical example of EO program.
        [start] > canonical
        """
    ).parsed()
).without("unused-void-attr").without("bipki").defects().forEach(System.out::println);
```
Output
```console
[canonical.canonical unused-void-attr (Single) WARNING]:9 The void attribute "start" is unused anywhere in the object "canonical"
```

2. The method takes strings, so if I pass an incorrect lint name, nothing happens and there's no feedback. I see two ways to address this:

    2.1. If we don't intend to support custom lints(as it currently stands - `Program` and `Source` don't accept `Lint` as an argument, even though `Lint` is a public interface), then we should make `Lint` `package-private` and generate enum of all existing lint names during build time(or write it manually). Then, `without()` would take this enum type, allowing compile-time checks to ensure the disables are valid

    2.2. If we do want to allow custom lints in the future, we should add a public constructor for `Program` and `Source` that accepts a list of custom linters. Then, `without()` should throw an exception if it tries to disable a non-existent lint

3. `without()` should accept one required argument followed by varargs, so that calling `.without()` with no arguments isn't possible.

**Comment by volodya-lombrozo** [comment]
@Marat-Tim you can use `.without("unused-void-attr", "bipki")`, no need to introduce mutable state for the `Source` object
